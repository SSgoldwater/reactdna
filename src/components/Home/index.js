import React, { Component } from 'react';
import styles from './styles';

class Home extends Component {
  constructor(props) {
    super(props);

    this.state = {
      count: 0
    };
  };

  _add = () => {
    this.setState({ count: this.state.count + 1 });
  }

  _subtract = () => {
    this.setState({ count: this.state.count - 1 });
  }

  render() {
    return (
      <div>
        Homepage
        <br />
        <p id={ `count` }>
          Count: { this.state.count }
        </p>
        <br />
        <button
          id={ `addButton` }
          onClick={ this._add }
        >Add</button>
        <button
          onClick={ this._subtract }
        >!Add</button>
      </div>
    );
  }
}

export default Home;
