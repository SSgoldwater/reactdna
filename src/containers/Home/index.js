import React, { Component } from 'react';
import { Container } from 'flux/utils';
import HomePage from 'components/HomePage';
import CounterStore from 'data/stores/CounterStore';

class Home extends Component {
  constructor(props) {
    super(props);

    const _counterState = CounterStore.getState();

    this.state = {
      count: _counterState.get(`count`)
    };
  }

  static getStores() {
    return [CounterStore];
  }

  static calculateState() {
    const _counterState = CounterStore.getState();

    return {
      count: _counterState.get(`count`)
    };
  }

  render() {
    return (
      <HomePage
        count={ this.state.count }
      />
    );
  }
};

const _Home = Container.create(Home);

export default _Home;
