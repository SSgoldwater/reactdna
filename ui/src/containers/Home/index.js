import React, { Component } from 'react';
import { Container } from 'flux/utils';
import HomePage from 'components/HomePage';
import TodoStore from 'data/stores/TodoStore';

class Home extends Component {
  constructor(props) {
    super(props);

    const todoState = TodoStore.getState();

    this.state = {
      inputValue: todoState.inputValue,
      todos: todoState.todos
    };
  }

  static getStores() {
    return [TodoStore];
  }

  static calculateState() {
    const todoState = TodoStore.getState();

    return {
      inputValue: todoState.inputValue,
      todos: todoState.todos
    };
  }

  render() {
    return (
      <HomePage
        todos={ this.state.todos }
        inputValue={ this.state.inputValue }
      />
    );
  }
};

const _Home = Container.create(Home);

export default _Home;
