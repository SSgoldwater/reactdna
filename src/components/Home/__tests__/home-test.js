import React from 'react';
import { shallow } from 'enzyme';
import Home from '../../Home';

test(`Home adds`, () => {
  const home = shallow(<Home />);

  expect(home.find(`#count`).text()).toMatchSnapshot();

  console.log(home.find(`#count`).text());

  home.find(`#addButton`).simulate('click');

  expect(home.find(`#count`).text()).toMatchSnapshot();

  console.log(home.find(`#count`).text());
});
