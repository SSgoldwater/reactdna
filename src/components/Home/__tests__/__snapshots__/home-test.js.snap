// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home adds 1`] = `"Count: 0"`;

exports[`Home adds 2`] = `"Count: 1"`;
